server/src/events.mjs

```diff
@@ -92,10 +93,79 @@ function sendEventListenerAck(ws, oMsg) {
   logger.debug(`Sent event listener ack message: ${ackMessage}`);
 }
 
+// sendEvent function to handle pre- post- method/event triggers
+function sendEvent(ws, method, result, msg, fSuccess, fErr, fFatalErr) {
+  try {
+    if ( !isRegisteredEventListener(method) ) {
+      fErr.call(null, method);
+
+    } else {
+       // Fire pre trigger if there is one for this method
+       if ( method in eventTriggers ) {
+        if ( 'pre' in eventTriggers[method] ) {
+          try {
+            const ctx = {
+              logger: logger,
+              setTimeout: setTimeout,
+              setInterval: setInterval
+            };
+            logger.debug(`Calling pre trigger for event ${method}`);
+            eventTriggers[method].pre.call(ctx);
+          } catch ( ex ) {
+            logger.error(`ERROR: Exception occurred while executing pre-trigger for ${method}; continuing`);
+          }
+        }
+      }
+
+      const id = getRegisteredEventListener(method);
+      let postResult;
+      
+      // Fire post trigger if there is one for this method
+      if ( method in eventTriggers ) {
+        if ( 'post' in eventTriggers[method] ) {
+          try {
+            const ctx = {
+              logger: logger,
+              setTimeout: setTimeout,
+              setInterval: setInterval,
+              result : result
```

Comment:
same comment as above.. we should send sendEvent in as well.

Reply:
added sendEvent inside ctx object